config/webhook/deployment_inject_patch.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: mutating-webhook-configuration
+webhooks:
+  - name: deployment.sidecar-injector.jaegertracing.io
+    objectSelector: # Skip resources with the name jaeger-operator
+      matchExpressions:
+        - key: app.kubernetes.io/name
+          operator: In
```

Comment:
Wouldn't you want `NotIn` here?

Reply:
For Deployments we offer a way to control auto-injection:

```yaml
  annotations:
    "sidecar.jaegertracing.io/inject": "true"
```

(See https://www.jaegertracing.io/docs/1.32/operator/#auto-injecting-jaeger-agent-sidecars )

Do we have a similar annotation on the Pod level?  I'd like to be able to attach

```yaml
  annotations:
    "sidecar.jaegertracing.io/inject": "false"
```

to pods (and pod-templates) that must not be attached to the sidecar.  Then I would like this webhook to check for that label, rather than only skipping mutation for the operator itself.

Reply:
https://github.com/jaegertracing/jaeger-operator/pull/1804 is closed.

The `failurePolicy: Ignore` is great but it should be possible for someone who doesn't want the sidecar to annotate their pods so that the webhook doesn't fire-then-fail.

For example, Istio's `MutatingWebhookConfiguration` uses

```YAML
  namespaceSelector:
    matchExpressions:
    - key: istio-injection
      operator: DoesNotExist
    - key: istio.io/rev
      operator: In
      values:
      - {{istio-instance}}
  objectSelector:
    matchExpressions:
    - key: "sidecar.istio.io/inject"
      operator: NotIn
      values:
      - "false"
```

Reply:
> https://github.com/jaegertracing/jaeger-operator/pull/1804 is closed.

Y, its not supported. We agreed in not extending the Jaeger-Operator functionality. Instead we focus on the [OTEL](https://github.com/open-telemetry/opentelemetry-operator).

> In v1.15+, webhooks may optionally limit which requests are intercepted based on the **labels** of the objects they would be sent, by specifying an objectSelector. 

[Istio](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#controlling-the-injection-policy) does this on a label basis using the [objectSelector](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-objectselector). Jaeger-Operator uses annotations instead. Unfortunately, these cannot be pre-filtered. That means all new deployments pass the webhook and [there is checked](https://github.com/jaegertracing/jaeger-operator/blob/main/controllers/appsv1/deployment_webhook.go#L101) for the corresponding annotation and value.